app/controllers/books_controller.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
+  before_action :find_single_book, only: %i[show edit update destroy]
+
+  def index
+    @books = Book.all
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def show
+    @reviews = Review.where(book_id: params[:id])
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def new
+    @book = Book.new
+  rescue StandardError => e
+    Rails.logger.error(e.message)
+  end
+
+  def create
+    @book = Book.new(book_params)
+    if @book.save
+      flash[:success] = 'Book create completed'
+      redirect_to @book
+    else
+      flash[:error] = 'Book create fail!'
+      render :index
+    end
+  rescue StandardError => e
+    Rails.logger.error(e.message)
```

Comment:
Normally method `save` is not raise any error. It will return boolean only.
Please check https://apidock.com/rails/ActiveRecord/Base/save

if you want to raise error from save, please use `save!`
Please check https://apidock.com/rails/ActiveRecord/Base/save%21